gadgets/snapshot_process/test/unit/snapshot_process_test.go

```diff
@@ -0,0 +1,143 @@
+)
+
+type ExpectedSnapshotProcessEvent struct {
+	Comm      string `json:"comm"`
+	Pid       int    `json:"pid"`
+	Tid       int    `json:"tid"`
+	Uid       uint32 `json:"uid"`
+	Gid       uint32 `json:"gid"`
+	ParentPid int    `json:"ppid"`
+}
+
+type testDef struct {
+	runnerConfig  *utilstest.RunnerConfig
+	generateEvent func() (int, error)
+	validateEvent func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedSnapshotProcessEvent) error
+}
+
+func TestSnapshotProcessGadget(t *testing.T) {
+	utilstest.RequireRoot(t)
+	runnerConfig := &utilstest.RunnerConfig{}
+	testCases := map[string]testDef{
+		"captures_events_with_no_filter": {
+			runnerConfig:  runnerConfig,
+			generateEvent: generateEvent,
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedSnapshotProcessEvent) error {
+				utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, sleepPid int) *ExpectedSnapshotProcessEvent {
+					return &ExpectedSnapshotProcessEvent{
+						Comm:      "sleep",
+						Pid:       sleepPid,
+						Tid:       sleepPid,
+						Uid:       0,
+						Gid:       0,
+						ParentPid: info.Pid,
+					}
+				})(t, info, fd, events)
+				return nil
+			},
+		},
+	}
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			capturedEvents := []ExpectedSnapshotProcessEvent{}
+			var processId int
+			runner := utilstest.NewRunnerWithTest(t, testCase.runnerConfig)
+			utilstest.RunWithRunner(t, runner, func() error {
+				pid, _ := testCase.generateEvent()
+				processId = pid
+				return nil
+			})
+			timeout := 5 * time.Second
+			const opPriority = 50000
+
+			gadgetOperator := simple.New("gadget",
+				simple.OnInit(func(gadgetCtx operators.GadgetContext) error {
+					for _, d := range gadgetCtx.GetDataSources() {
+						jsonFormatter, _ := igjson.New(d)
+						d.Subscribe(func(source datasource.DataSource, data datasource.Data) error {
+							event := &ExpectedSnapshotProcessEvent{}
+							jsonOutput := jsonFormatter.Marshal(data)
+							err := json.Unmarshal(jsonOutput, event)
+							require.NoError(t, err, "unmarshalling event")
+
+							// for current reference of the event
+							fmt.Println("Captured data: ", string(jsonOutput))
+
+							capturedEvents = append(capturedEvents, *event)
+
+							return nil
+						}, opPriority)
+
+					}
+					return nil
+				}))
+
+			gadgetCtx := gadgetcontext.New(
+				context.Background(),
+				// Use the snapshot_process gadget, this part will be made more modular in the future
+				"ghcr.io/inspektor-gadget/gadget/snapshot_process:latest",
+				gadgetcontext.WithDataOperators(
+					ocihandler.OciHandler,
+					gadgetOperator,
+				),
+				gadgetcontext.WithTimeout(timeout),
+			)
+			runtime := local.New()
+			err := runtime.Init(nil)
+			require.NoError(t, err, "initializing runtime")
+			params := map[string]string{
+				"operator.oci.ebpf.uid": "0",
+			}
```

Comment:
nit: This gadget doesn't support this parameter. Check supported parameters in https://www.inspektor-gadget.io/docs/latest/gadgets/snapshot_process#flags. Or by looking at GADGET_PARAM in the code.